tests/system/Honeypot/HoneypotTest.php

```diff
@@ -66,16 +71,35 @@ public function testAttachHoneypotAndContainer()
     {
         $this->response->setBody('<form></form>');
         $this->honeypot->attachHoneypot($this->response);
-        $expected = '<form><div style="display:none"><label>Fill This Field</label><input type="text" name="honeypot" value=""/></div></form>';
+        $expected = '<form><div style="display:none"><label>Fill This Field</label><input type="text" name="honeypot" value=""></div></form>';
         $this->assertSame($expected, $this->response->getBody());
 
         $this->config->container = '<div class="hidden">{template}</div>';
         $this->response->setBody('<form></form>');
         $this->honeypot->attachHoneypot($this->response);
-        $expected = '<form><div class="hidden"><label>Fill This Field</label><input type="text" name="honeypot" value=""/></div></form>';
+        $expected = '<form><div class="hidden"><label>Fill This Field</label><input type="text" name="honeypot" value=""></div></form>';
         $this->assertSame($expected, $this->response->getBody());
     }
 
+    public function testAttachHoneypotAndContainerWithCSP()
+    {
+        $this->resetServices();
+
+        $config             = new App();
+        $config->CSPEnabled = true;
+        Factories::injectMock('config', 'App', $config);
```

Comment:
Because `CodeIgniter\HTTP\ContentSecurityPolicy` depends on `config('App')`,
and `Response` pulls `Services::csp()`, we must inject `Config\App` into the Factories.

Reply:
Thanks for the explanation! We can also just use the factory version here:
```php
config('App')->CSPEnabled = true;
```

But nothing wrong with the explicit version.

Reply:
Factories are reset each time during the test, so there is probably no problem with that way.
However, I do not think it is healthy for most framework class to depends on Factories (`config()`).

It's not easy to get rid of this dependency now, but I would like to reduce it if possible.